### PR TITLE
Add test case for DateUtils

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #CHT

**Type**
Choose one: Testing

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt [hunk 1]
```diff
@@ -38,6 +39,27 @@ class DateUtilsTest {
     Assert.assertEquals("3-22-2020", finalDate)
   }
 
+  @Test
+  fun `addDays() should return same date when given initialDate and pattern`() {
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:43]: Could we change this to something to reflect that zero days were added? e.g.

`addDays should return same date when default 0 days are added`

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt [hunk 2]
```diff
@@ -38,6 +39,27 @@ class DateUtilsTest {
...
+  @Test
+  fun `hasPastDays should return true when given initialDate`() {
+    val initialDate = DateTimeType("2022-01-11T15:30:10.222")
+    val hasPastDays = DateUtils.hasPastDays(initialDate)
+    Assert.assertTrue(hasPastDays)
+  }
+
   @Test
   fun `String#getDate() should return correct date when given string format expected`() {
     Assert.assertEquals("2020-03-10".getDate("yyyy-MM-dd").makeItReadable(), "10-Mar-2020")
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:65]: The **Expected** value should always be the first parameter. Please switch these around

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt [hunk 3]
```diff
@@ -38,6 +39,27 @@ class DateUtilsTest {
...
+
+  @Test
+  fun `hasPastDays should return true when given initialDate and days`() {
+    val initialDate = DateTimeType("2022-01-11T15:30:10.222")
+    val hasPastDays = DateUtils.hasPastDays(initialDate, -3)
+    Assert.assertTrue(hasPastDays)
+  }
+
+  @Test
+  fun `hasPastDays should return true when given initialDate`() {
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:57]: We need to re-word these test case name as well, it is hard to tell exactly which conditions are under test

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt [hunk 4]
```diff
@@ -38,6 +39,27 @@ class DateUtilsTest {
...
 
+  @Test
+  fun `addDays() should return same date when given initialDate and pattern`() {
+    val finalDate =
+      DateUtils.addDays("2020-03-10 01:23:00 AM", dateTimeFormat = "yyyy-MM-dd h:mm:ss a")
+    Assert.assertEquals("3-10-2020", finalDate)
+  }
+
+  @Test
+  fun `hasPastDays should return true when given initialDate and days`() {
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:50]: Can we re-word this to capture conditions under test and expected output?

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt (hunk 1 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:43]: Yeah sure

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt (hunk 2 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:65]: Okay, thanks, I missed that one

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt (hunk 4 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:50]: Is this what you have in mind?
`hasPastDays() should return true when given valid initialDate and has negative days`

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt (hunk 3 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:57]: How about this?
`hasPastDays() should return true when given valid initialDate and has default 0 days`

Comment (ndegwamartin) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:57]: Yeah that can work

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt (hunk 4 above)
Comment (ndegwamartin) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:50]: maybe something like `hasPastDays() should return true when given  date is 3 days in the past`

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt (hunk 3 above)
Comment (ndegwamartin) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:57]: By looking at the test case name, it should be easy to tell what function the method is doing (and condition we are testing) without looking at the code

Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:57]: Oh I see, yeah, thanks, I will keep that in mind

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt (hunk 4 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/DateUtilsTest.kt:50]: Yeah, that makes sense

**Review by pld — APPROVED** [review]

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**